app/models/manageiq/providers/lenovo/physical_infra_manager/parser/network_device_parser.rb

```diff
@@ -90,7 +90,7 @@ def parse_child_devices(device, all_devices)
 
       def parse_ports(port)
         port_info = port["portInfo"]
-        physical_ports = port_info["physicalPorts"]
+        physical_ports = port_info&.try(:[], 'physicalPorts')
```

Comment:
I believe that a `try` call here is enough, you can remove the `&`.

Reply:
That's correct, or you can use `port_info&.dig('physicalPorts')`...up to you but I've never liked the way `.try` looks with a hash dereference

Reply:
Cool, .dig seems cleaner to me